subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/AuthenticationSupporter.java

```diff
@@ -135,6 +136,8 @@ private void populateAuthenticationCredentials() {
             return Cast.uncheckedCast(DefaultPasswordCredentials.class);
         } else if (publicType == AwsCredentials.class) {
             return Cast.uncheckedCast(DefaultAwsCredentials.class);
+        } else if (publicType == HttpHeaderCredentials.class) {
+            return Cast.uncheckedCast(DefaultHttpHeaderCredentials.class);
         } else {
             throw new IllegalArgumentException(String.format("Unknown credentials type: '%s' (supported types: %s and %s).", publicType.getName(), PasswordCredentials.class.getName(), AwsCredentials.class.getName()));
```

Comment:
`HttpHeaderCredentials` is missing from the exception message.

Reply:
Thanks, fixed.